panther_battery/include/panther_battery/adc_data_reader.hpp

```diff
@@ -0,0 +1,50 @@
+#ifndef PANTHER_BATTERY_ADC_DATA_READER_HPP_
+#define PANTHER_BATTERY_ADC_DATA_READER_HPP_
+
+#include <math.h>
+
+#include <filesystem>
+#include <fstream>
+#include <map>
+#include <stdexcept>
+#include <string>
+
+namespace panther_battery
+{
+
+class ADCDataReader
+{
+public:
+  ADCDataReader(const std::string & device_path) : device_path_(device_path) {}
+
+  float GetADCMeasurement(const std::string & file, const float & offset, const float & LSB);
+
+private:
+  const std::string device_path_;
+
+  int ReadFile(std::string file_path);
+};
+
+inline float ADCDataReader::GetADCMeasurement(
+  const std::string & file, const float & offset, const float & LSB)
```

Comment:
```suggestion
  const std::string & file, const float offset, const float LSB)
```
I don't think passing basic types as reference is a good idea. For objects it makes sense, but for float I do not second this

Reply:
Done